mono/mini/mini.c

```diff
@@ -4168,6 +4168,11 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 		return NULL;
 	}
 
+	if (mono_class_is_gtd (method->klass)) {
+		mono_error_set_exception_instance (error, mono_get_exception_invalid_operation ("Could not execute the method because the containing type is not fully instantiated."));
```

Comment:
You can use `mono_error_set_invalid_operation()`

Reply:
Does it make sense to check for `mono_class_is_open_constructed_type` rather than just for a GTD?

Also what about if we're attempting to call a generic method definition?

Reply:
This should be !cfg->compile_aot &&... The aot compiler passes all kinds of generic methods to the JIT.

Reply:
@lambdageek I wasn't able to create a delegate with generic method definition, nor to invoke it some other way